crates/library/std/src/buf.fe

```diff
@@ -47,7 +47,7 @@ pub struct MemoryBuffer {
 
     pub fn new(len: u256) -> Self {
         unsafe {
-            return MemoryBuffer(offset: alloc(len), len)
+            return MemoryBuffer(offset: alloc(len: len + 31), len)
```

Comment:
Couldn't we check if `len` is a multiple of `32` and then not allocate 31 superfluous bytes?

Reply:
it's actually not a multiple of 32 thing, as the buffer can be written to with anywhere between 1-32 bytes. problem is that we can only write 1 or 32 bytes at the evm level, so it needs to be padded `32 - <smallest write size>` where the "smallest write size" is the the smallest number of bytes that we write at the buffer level using the 32 byte write operation

Im realizing now that since this "smallest write size" is actually 2 bytes for `u16`, the number should isntead be 30

Reply:
Uuuh..I think I finally get it!